joomla/joomla-cms
# [4.0] Normalise Featured Contact List

### Summary of Changes
Follow up on https://github.com/joomla/joomla-cms/pull/30813 and https://github.com/joomla/joomla-cms/pull/30859
Warning: these 2 PR have to be present in your test site to test the Clear button.

**npm ci has to be applied** 

Changes:
Normalises the table display
Adds a Title Search Filter, Filter and Clear buttons.
a11y legend and scope added
Modified the search placeholder (it is also used in Contact List but I forgot to modify it there)
Used Title instead of Name for column heading, as in Contact list

### Testing Instructions
Create some contacts and set them to featured.
Create a Featured Contacts menu item.
Display in frontend.

After patch, test that the Filter and the Clear button work as should


### Actual result BEFORE applying this Pull Request

<img width="840" alt="Screen Shot 2020-10-02 at 11 50 02" src="https://user-images.githubusercontent.com/869724/94913860-4c8a1000-04aa-11eb-8c09-b7fce1f57cc4.png">


### Expected result AFTER applying this Pull Request

<img width="813" alt="Screen Shot 2020-10-02 at 12 14 04" src="https://user-images.githubusercontent.com/869724/94913921-64619400-04aa-11eb-81fb-d8488c3b1511.png">


### NOTE
No idea why we display the NUM in the first column. 
If nobody knows why, we can get rid of it.

Meaanwhile, I kept in the code
```
					<tr class="<?php echo ($i % 2) ? 'odd' : 'even'; ?>" itemscope itemtype="https://schema.org/Person">
						<td class="item-num">
							<?php echo $i; ?>
						</td>
```
